### ISSUE TITLE
`assertion failed: !ptr.is_null()` when calling `Native::surface`

### ISSUE THREAD

**Issue by awused** [gtk-rs/gtk4-rs#1615]
**Bug description**

Calling `window.native().map(|n| n.surface()))` before a Window has been presented crashes with a null pointer assertion.

It should not crash, it should return an Optional<Surface> instead.

Looking at the clipboard example, this patch will cause it to crash.

```
diff --git a/examples/clipboard/main.rs b/examples/clipboard/main.rs
index 7f6c3b759a..5cc99cfba7 100644
--- a/examples/clipboard/main.rs
+++ b/examples/clipboard/main.rs
@@ -126,5 +126,6 @@ fn build_ui(application: &gtk::Application) {
     container.append(&texture_container);

     window.set_child(Some(&container));
+    println!("{:?}", window.native().map(|n| n.surface())));
     window.present();
 }
```

**Comment by sdroege** [comment]
`gtk_native_get_surface()` is not marked as `nullable` in the C API, so either it is a bug that `NULL` is returned at all or the annotation is missing. The latter is the case, and it's fixed here: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6696

We should get the update with the next release here.

**Comment by bilelmoussaoui** [comment]
it is already marked as nullable here https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/prelude/trait.NativeExt.html#method.surface
so probably using an older gtk4-rs

**Comment by bilelmoussaoui** [comment]
Any updates here ?  I feel like there is nothing to be done here.

**Comment by awused** [comment]
Well, I can't exactly test on the newest version when it doesn't work. You can close it as likely to be fixed, or I'll update it once 0.8.1 comes out.

**Comment by bilelmoussaoui** [comment]
0.8.1 is out
